apps/devhub/tasks.py

```diff
@@ -32,21 +32,23 @@
 log = logging.getLogger('z.devhub.task')
 
 
-@task
+@task(soft_time_limit=settings.VALIDATOR_TIMEOUT)
 @write
 def validator(upload_id, listed=True, **kw):
     if not settings.VALIDATE_ADDONS:
         return None
+
     log.info('VALIDATING: %s' % upload_id)
     upload = FileUpload.objects.using('default').get(pk=upload_id)
     try:
         result = run_validator(upload.path, listed=listed)
-    except:
+    except Exception:
```

Comment:
No. A bare except catches things which inherit from `BaseException`, but not `Exception`, which generally shouldn't be caught by an exception handler. This has become a common anti-pattern: http://stackoverflow.com/questions/18982610/difference-between-except-and-except-exception-as-e-in-python